### ISSUE TITLE
JUnit  formater doesn't report Scenario outline example with -x argument

### ISSUE THREAD

**Issue by floreal** [cucumber/cucumber#124]
Hello,
I noticed a little annoying bug. consider my Rakefile:

```ruby
require 'rubygems'
require 'rake'
require 'cucumber/rake/task'

# some other tasks

namespace :cucumber do
  Cucumber::Rake::Task.new(:report) do |t|
    t.cucumber_opts = "-x -p #{browser} -f junit --out report -f pretty --no-color"
  end
  
  Cucumber::Rake::Task.new(:pretty) do |t|
    t.cucumber_opts = "-x -p #{browser}"
  end
end
```
The cucumber proffiles only define environment variable to set selenium url and desired browser capabilities

When i execure rake cucumber:report, only Scenarios are reported to the JUnit result report, Scenario Outlines examples aren't.
When I remove the -x flag they are reported.

**Comment by mattwynne** [comment]
@brasmusson @richarda either of you fancy a crack at this one?

**Comment by richarda** [comment]
Sure!

**Comment by tooky** [comment]
Closed in favour of #808
